File: app/src/main/java/com/alexandr7035/txnotes/db/NotesDao.java
package com.alexandr7035.txnotes.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface NotesDao {


    @Query("SELECT COUNT(*) FROM notes")
    LiveData<Integer> getNotesCount();

    @Query("SELECT * FROM notes ORDER BY id")
    LiveData<List<NoteEntity>> getAllNotes();

    @Query("SELECT * FROM notes WHERE id = (:id)")
    NoteEntity getNoteById(long id);

    @Insert
    long addNote(NoteEntity note);

    @Update
    void updateNote(NoteEntity noteEntity);

    @Delete
    void deleteNote(NoteEntity noteEntity);
}

File: data/src/main/java/by/alexandr7035/data/local/NotesDao.kt
package by.alexandr7035.data.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface NotesDao {
    @Insert
    suspend fun saveNote(note: NoteEntity)

    @Query("SELECT * FROM notes ORDER by id DESC")
    suspend fun getNotesList(): List<NoteEntity>
}